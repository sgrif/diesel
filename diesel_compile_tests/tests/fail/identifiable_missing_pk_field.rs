#[macro_use]
extern crate diesel;

table! {
    foo {
        id -> Integer,
    }
}

#[derive(Identifiable)]
#[diesel(table_name = foo)]
struct Foo1 {}

#[derive(Identifiable)]
#[diesel(table_name = foo)]
struct Foo2 {
    #[diesel(column_name = foo)]
    id: i32,
}

#[derive(Identifiable)]
#[diesel(primary_key(bar))]
#[diesel(table_name = foo)]
struct Foo3 {}

#[derive(Identifiable)]
#[diesel(primary_key(baz))]
#[diesel(table_name = foo)]
struct Foo4 {
    #[diesel(column_name = bar)]
    baz: i32,
}

#[derive(Identifiable)]
#[diesel(primary_key(foo, bar))]
#[diesel(table_name = foo)]
struct Foo5 {
    foo: i32,
}

#[derive(Identifiable)]
#[diesel(primary_key(foo, bar))]
#[diesel(table_name = foo)]
struct Foo6 {
    foo: i32,
    #[diesel(column_name = baz)]
    bar: i32,
}

fn main() {}
