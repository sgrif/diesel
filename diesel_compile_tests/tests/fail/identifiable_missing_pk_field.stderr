error: No field with column name id
  --> $DIR/identifiable_missing_pk_field.rs:10:10
   |
10 | #[derive(Identifiable)]
   |          ^^^^^^^^^^^^
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: No field with column name id
  --> $DIR/identifiable_missing_pk_field.rs:14:10
   |
14 | #[derive(Identifiable)]
   |          ^^^^^^^^^^^^
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: No field with column name bar
  --> $DIR/identifiable_missing_pk_field.rs:22:22
   |
22 | #[diesel(primary_key(bar))]
   |                      ^^^

error: No field with column name baz
  --> $DIR/identifiable_missing_pk_field.rs:27:22
   |
27 | #[diesel(primary_key(baz))]
   |                      ^^^

error: No field with column name bar
  --> $DIR/identifiable_missing_pk_field.rs:35:27
   |
35 | #[diesel(primary_key(foo, bar))]
   |                           ^^^

error: No field with column name bar
  --> $DIR/identifiable_missing_pk_field.rs:42:27
   |
42 | #[diesel(primary_key(foo, bar))]
   |                           ^^^
